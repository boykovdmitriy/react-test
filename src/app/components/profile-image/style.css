.image_preview {
    max-width: 3rem;
}