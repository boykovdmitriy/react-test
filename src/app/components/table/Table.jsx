import React from 'react';
import style from './table.css';

export const Table = ({children, headers}) => (
    <table className={style.table}>
        <tbody>
        <tr >
            {headers.map(x => (
                <th className={style.header} key={x}>{x}</th>
            ))}
        </tr>
        {children}
        </tbody>
    </table>
);