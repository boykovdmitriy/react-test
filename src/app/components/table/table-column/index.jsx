import React from 'react';
import style from '../table.css'

export const TableColumn = ({children}) => (
    <td className={style.column}>
        {children}
    </td>
);