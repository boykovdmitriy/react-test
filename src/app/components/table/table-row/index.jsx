import React from 'react';
import style from '../table.css'

export const TableRow = ({children}) => (
    <tr className={style.row}>
        {children}
    </tr>
);