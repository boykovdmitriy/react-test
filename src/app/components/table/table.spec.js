import React from 'react';
import {Table} from './Table'
import style from './table.css';
import {TableRow} from "./table-row";
import {TableColumn} from "./table-column";

describe('table component', () => {
    it('render empty table component', () => {
        const wrapper = shallow(
            <Table/>
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render table with header', () => {
        const headers = ['name', 'website', 'age'];
        const wrapper = shallow(
            <Table headers={headers}/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(
            wrapper.contains(
                <tr>
                    {headers.map(x => (
                        <th className={style.header} key={x}>{x}</th>
                    ))}
                </tr>
            )).toEqual(true);
    });
    it('render table with rows and columns', () => {
        const values = [{
            id: 1,
            columnName: 'first'
        }, {
            id: 2,
            columnName: 'second'
        },];
        const wrapper = shallow(
            <Table>
                {
                    values.map(x => (
                        <TableRow key={x.id}>
                            <TableColumn>{x.columnName}</TableColumn>
                        </TableRow>
                    ))
                }
            </Table>
        );
        expect(wrapper).toMatchSnapshot();
        expect(
            wrapper.contains(
                values.map(x => (
                    <TableRow key={x.id}>
                        <TableColumn>{x.columnName}</TableColumn>
                    </TableRow>
                ))
            )).toEqual(true);
    });
    it('render row with column', () => {
        const wrapper = shallow(
            <TableRow>
                <TableColumn>column</TableColumn>
            </TableRow>
        );
        expect(wrapper).toMatchSnapshot();
        expect(
            wrapper.contains(
                <tr className={style.row}><TableColumn>column</TableColumn></tr>
            )
        ).toEqual(true);
    });
    it('render column', () => {
        const wrapper = shallow(
            <TableColumn>column</TableColumn>
        );
        expect(wrapper).toMatchSnapshot();
        expect(
            wrapper.contains(
                <td className={style.column}>column</td>
            )
        ).toEqual(true);
    });
});