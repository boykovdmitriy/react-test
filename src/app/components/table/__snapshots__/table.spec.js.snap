// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table component render Test component with items 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Name
      </th>
      <th>
        Website
      </th>
      <th>
        Image
      </th>
    </tr>
    <TableItem
      key="1"
      salon={
        Object {
          "id": 1,
          "name": "first",
          "profile_image_urls": Object {
            "original": "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/original/1.jpg?1448538378",
            "thumb": "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/thumb/1.jpg?1448538378",
          },
          "website": "http://mohr.co",
        }
      }
    />
    <TableItem
      key="1"
      salon={
        Object {
          "id": 1,
          "name": "second",
          "profile_image_urls": Object {
            "original": "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/original/1.jpg?1448538378",
            "thumb": "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/thumb/1.jpg?1448538378",
          },
          "website": "http://mohr.co",
        }
      }
    />
  </tbody>
</table>
`;
